use upgraded instance types in default-for-karpenter

demonCoder95 opened the pull request:
We decided to perform instance type upgrades:

| Instance Type | Replacement |
|---------------|-------------|
| c5            | c6i         |
| m5            | m6i         |
| r5            | r6i         |
| ~c6g~           | ~c7g~         |

For the `default-for-karpenter` type, we can simply drop the older types as we already have the new ones in the list. `c6g` is not in the list so no replacement is needed.

demonCoder95 committed:
use upgraded instance types in default-for-karpenter

demonCoder95 commented:
👍

katyanna commented:
| Instance Type | Replacement |
|---------------|-------------|
| c5            | c6i         |
| m5            | m6i         |
| c6g           | c7g         |

quick review of the table in the description, is that correct?

katyanna commented:
| Instance Type | Replacement |
|---------------|-------------|
| c5            | c6i         |
| m5            | m6i         |
| c6g           | c7g         |

quick review of the table in the description; similar to my comment on the e2e PR, I think there was a typo on the m5 replacement and there's no c6 changes in this PR. Is that correct?

demonCoder95 commented:
I fixed the typo, thanks for highlighting it 🙂 Yes, that's correct. I copied the table from the internal issue we have for the replacement strategy, I copied the typo over 🙈 I fixed it now.

mikkeloscar commented:
The PR looks ok, but the description also mentions c6g/c7g which should not (and is not) included.

Those should not be included as there would be the risk of spinning up an ARM node for pods with a container image only working on amd64.

I crossed those out in the description.

mikkeloscar commented:
:+1:
